src/api/topics/topics.controller.ts

```diff
@@ -0,0 +1,44 @@
+import {
+  Controller,
+  Get,
+  Post,
+  Body,
+  Patch,
+  Param,
+  Delete,
+} from '@nestjs/common';
+
+import { CreateTopicDto } from '@api/topics/dto/create-topic.dto';
+import { UpdateTopicDto } from '@api/topics/dto/update-topic.dto';
+import { TopicEntity } from '@api/topics/topic.entity';
+import { TopicsService } from '@api/topics/topics.service';
+
+@Controller('topics')
+export class TopicsController {
+  constructor(private readonly topicsService: TopicsService) {}
+
+  @Post()
+  create(@Body() createTopicDto: CreateTopicDto) {
+    return this.topicsService.create(createTopicDto);
+  }
+
+  @Get()
+  findAll(): Promise<TopicEntity[]> {
+    return this.topicsService.findAll();
+  }
+
+  @Get(':id')
+  findOne(@Param('id') id: string): Promise<TopicEntity | null> {
```

Comment:
Here you could use the built-in nest pipe to ensure the param id is an integer
```suggestion
  findOne(@Param('id', ParseIntPipe) id: number): Promise<TopicEntity | null> {
```